operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -870,6 +879,35 @@ func (r *RedpandaReconciler) reconcileDelete(ctx context.Context, rp *redpandav1
 	return nil
 }
 
+func (r *RedpandaReconciler) deleteHelmChart(ctx context.Context, rp *redpandav1alpha2.Redpanda) error {
+	// When HelmRelease is suspended it will not delete child resource HelmChart. In this function there is attempt
+	// to delete HelmChart custom resource.
+	// Reference:
+	// https://github.com/fluxcd/helm-controller/blob/2d335f2aa0e2e0df2a631ebf19394aed07c556f3/internal/reconcile/helmchart_template.go#L163-L166
+	// https://github.com/fluxcd/helm-controller/blob/2d335f2aa0e2e0df2a631ebf19394aed07c556f3/internal/controller/helmrelease_controller.go#L375-L388
+	namespacedName := client.ObjectKey{Namespace: rp.Namespace, Name: rp.Namespace + "-" + rp.Name}
+	var chart sourcev1.HelmChart
+	err := r.Client.Get(ctx, namespacedName, &chart)
+	if err != nil && !apierrors.IsNotFound(err) {
+		// Return error to retry until we succeed.
+		return fmt.Errorf("failed to get flux HelmChart '%s': %w", namespacedName.Name, err)
+	} else if err == nil {
+		if controllerutil.ContainsFinalizer(&chart, FluxFinalizerKey) {
```

Comment:
I think you could also specify `Force` as part of the delete options rather than manually removing this finalizer. No preference though.

Reply:
I will check with the `Force` option.